### PR TITLE
🐛 add wlp annotation if necessary

### PR DESCRIPTION
**Pull request by elgnay**
## Related issue(s)

Fixes #

### CODE REVIEW

File: pkg/operator/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go [hunk 1]
```diff
@@ -333,20 +335,39 @@ func getManagedKubeConfig(ctx context.Context, kubeClient kubernetes.Interface,
...
+func ensureNamespaceWithAnnotation(ctx context.Context, kubeClient kubernetes.Interface, namespace, annotationName,
+	annotationValue string) error {
+	ns, err := kubeClient.CoreV1().Namespaces().Get(ctx, namespace, metav1.GetOptions{})
 	if errors.IsNotFound(err) {
 		_, createErr := kubeClient.CoreV1().Namespaces().Create(ctx, &corev1.Namespace{
 			ObjectMeta: metav1.ObjectMeta{
 				Name: namespace,
 				Annotations: map[string]string{
-					"workload.openshift.io/allowed": "management",
+					annotationName: annotationValue,
```
Comment (qiujian16) [inline pkg/operator/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go:348]: why it is not hard coded?

File: pkg/operator/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go [hunk 2]
```diff
@@ -333,20 +335,39 @@ func getManagedKubeConfig(ctx context.Context, kubeClient kubernetes.Interface,
...
 }
 
-// ensureAgentNamespace create agent namespace if it is not exist
-func ensureAgentNamespace(ctx context.Context, kubeClient kubernetes.Interface, namespace string) error {
-	_, err := kubeClient.CoreV1().Namespaces().Get(ctx, namespace, metav1.GetOptions{})
+// ensureNamespaceWithAnnotation create the namespace if necessary and make sure it has the
+// given annotation
+func ensureNamespaceWithAnnotation(ctx context.Context, kubeClient kubernetes.Interface, namespace, annotationName,
+	annotationValue string) error {
+	ns, err := kubeClient.CoreV1().Namespaces().Get(ctx, namespace, metav1.GetOptions{})
```
Comment (qiujian16) [inline pkg/operator/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go:342]: hrm, if it is not create only, maybe you can just use use applyNamespace in library go

File: pkg/operator/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go (hunk 1 above)
Comment (elgnay) [reply pkg/operator/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go]: Use hard coded string instead.

File: pkg/operator/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go (hunk 2 above)
Comment (elgnay) [reply pkg/operator/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go]: Use `applyNamespace` from the openshift library-go instead.

**Review by qiujian16 — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**